src/library_sockfs.js

```diff
@@ -48,8 +48,8 @@ mergeInto(LibraryManager.library, {
     createSocket: function(family, type, protocol) {
       type &= ~{{{ cDefine('SOCK_CLOEXEC') | cDefine('SOCK_NONBLOCK') }}}; // Some applications may pass it; it makes no sense for a single process.
       var streaming = type == {{{ cDefine('SOCK_STREAM') }}};
-      if (protocol) {
-        assert(streaming == (protocol == {{{ cDefine('IPPROTO_TCP') }}})); // if SOCK_STREAM, must be tcp
+      if (streaming && protocol && protocol != {{{ cDefine('IPPROTO_TCP') }}}) {
+        throw new FS.ErrnoError({{{ cDefine('EPROTONOSUPPORT') }}}); // if SOCK_STREAM, must be tcp or 0.
```

Comment:
Is it really ok to be 0?

Reply:
In my experience it is very common to use ``socket(AF_INET, SOCK_STREAM, 0)`` to create a IPv4 TCP socket. I have rarely seen ``IPPROTO_TCP``. It works fine on Linux, FreeBSD, macOS, and Windows.

Reply:
Ah yes. I see.  From the man page: 

```
Normally only  a  single  protocol exists to support a particular socket type within a given
protocol family, in which case protocol can be specified as 0.
```